assets/src/scripts/lib/redux.js

```diff
@@ -72,16 +72,32 @@ export function provide(store) {
  * @param  {Object} selector Source selector for options
  * @return {Function}        D3 callback
  */
-
 export function link(func, selector) {
     let currentOptions = null;
-    let retval = null;
+    let context = null;
     return connect(function (selection, state) {
         let nextOptions = selector(state);
         if (currentOptions !== nextOptions) {
             currentOptions = nextOptions;
-            retval = func.call(null, selection, currentOptions);
+            context = func.call(null, selection, currentOptions, context);
```

Comment:
Yes tests would be good for that. What's going on here is the function provided to `link` will get passed the return value of its previous call. I used it in `createFocusCircles` to handle tracking the container the circles go into, but I think it may be useful in other cases.